src/network/NetworkController.ts

```diff
@@ -126,10 +207,14 @@ export class NetworkController extends BaseController<
       default:
         throw new Error(`Unrecognized network type: '${type}'`);
     }
+    this.getEIP1559Compatibility();
```

Comment:
Nit: Deceptively, this function does asynchronous work and returns a Promise, despite not being marked as `async`.

Calling this here to mirror the old behavior of this being in the constructor makes sense. But this is worth a comment to explain why we're doing this. e.g. something like `// asynchronously initialize the `isEIP1559Compatible` property`

It probably makes sense to block on this here, now that it's not in the constructor anymore. But that'd be an additional breaking change. We could do that here or later, either way is fine.

Reply:
Ah yes, agreed. I started a second branch that tries to make network controller methods async, but it broke a whole lot more tests: https://github.com/MetaMask/controllers/compare/refactor-network-controller-v2...network-controller-overhaul